app/controllers/two_factor_authentication/webauthn_verification_controller.rb

```diff
@@ -33,6 +34,22 @@ def confirm
 
     private
 
+    def check_if_device_supports_platform_auth
+      return unless user_session.has_key?(:platform_authenticator_available)
+      if platform_authenticator? && !device_supports_webauthn_platform?
+        redirect_to login_two_factor_options_url
+      end
+    end
+
+    def device_supports_webauthn_platform?
+      supports_webauthn_platform = false
+      if user_session[:platform_authenticator_available]
+        supports_webauthn_platform = true
+      end
+      user_session.delete(:platform_authenticator_available)
+      supports_webauthn_platform
```

Comment:
Could this be collapsed to... ?

```suggestion
      !!user_session.delete(:platform_authenticator_available)
```

Or:

```suggestion
      user_session.delete(:platform_authenticator_available) == true
```